### DISCUSSION TITLE
New Image 2.4.3 has no aws commands?

### DISCUSSION THREAD

**Discussion by Mais316** [argoproj/argo-cd#9887] [Q&A] [answered]
Is awscli removed from last image 2.4.3?

**Comment by crenshaw-dev** [comment]
@Mais316 it has been removed for a while. The reasons are documented here: https://github.com/argoproj/argo-cd/pull/8032

**Comment by Mais316** [comment]
We used awscli to copy backup to s3 bucket, building new image on top of 2.4.3 is failing


```sh
FROM argoproj/argocd:v2.4.3

# Switch to root for the ability to perform install
USER root


# (e.g. curl, awscli, gpg, sops)
RUN apt-get update && \
    apt-get install -y \
        curl \
        awscli \
        gpg && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
    curl -o /usr/local/bin/sops -L https://github.com/mozilla/sops/releases/download/3.2.0/sops-3.2.0.linux && \
    chmod +x /usr/local/bin/sops

# Switch back to non-root user
USER 999

```

Error

Reading package lists...
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/ca
E: Sub-process returned an error code

**Reply by crenshaw-dev** [reply]
Can you bump up the log level of `apt-get` to see if there are any more details on that error message? The fact that it's having trouble removing files makes me think there's some kind of path or permissions issue.

**Comment by Mais316** [comment, marked as answer]
I fixed the issue by using 2 containers to do the work one for argocd to take the backup and another one to copy it to s3, took the idea from https://github.com/argoproj/argo-cd/blob/99583c74c31850be24a5c6c5b54e86e14af0d5c5/docs/operator-manual/custom_tools.md 

@crenshaw-dev thanks for your help :).
